proto/documents/v1alpha/web_publishing.proto

```diff
@@ -0,0 +1,330 @@
+  string document_id = 1;
+
+  // Optional. Specific version of a document.
+  string version = 2;
+}
+
+// Response with a list of publication records among the web sites.
+message ListWebPublicationRecordsResponse {
+  repeated WebPublicationRecord publications = 1;
+}
+
+// Local site configuration.
+message SiteConfig {
+  // Hostname of the site.
+  string hostname = 1;
+
+  // Our role on this site.
+  Member.Role role = 2;
+}
+
+// Member of the web site.
+message Member {
+  // Roles a member of the Mintter Site can have.
+  enum Role {
+    // Invalid default value.
+    ROLE_UNSPECIFIED = 0;
+
+    // Currently there can only be one owner of the site.
+    OWNER = 1;
+
+    // Editors are allowed to push content to the site.
+    EDITOR = 2;
+  }
+
+  // Mintter Account ID of the member.
+  string account_id = 1;
+
+  // Member's role on the site.
+  Role role = 2;
+}
+
+// Information about a Mintter Document published on a Mintter Site.
+// One Document ID can be published with different versions and/or
+// under different paths on a Mintter Web Site, so it can have
+// multiple of these records per site.
+message WebPublicationRecord {
+  string document_id = 1;
+
+  string version = 2;
+
+  string hostname = 3;
+
+  string path = 4;
+}
+
+// API exposed by the remote server that allows
+// publishing Mintter Documents to serve them on the Web.
+// The server implementing this API is responsible for authenticating
+// callers for its administrative RPCs.
+// This API is also exposed by the local application backend, which would act
+// as a proxy to the remote web server, handling the authentication with its device private key.
+// The local application backend would require a special metadata header specifying the hostname
+// of the Mintter Site to proxy the request to.
+//
+// TODO(burdiyan): Document name of the required metadata header.
+service WebSite {
+  // Creates a new invite token for registering a new member.
+  // Protected. Must require authentication.
+  rpc CreateInviteToken(CreateInviteTokenRequest) returns (InviteToken);
+
+  // Redeems a previously created invite token to register a new member.
+  rpc RedeemInviteToken(RedeemInviteTokenRequest) returns (RedeemInviteTokenResponse);
+
+  // Gets public-facing site information.
+  rpc GetSiteInfo(GetSiteInfoRequest) returns (SiteInfo);
+
+  // Updates public-facing site information. Doesn't support partial updates,
+  // hence all the fields must be provided.
+  // Protected. Must require authentication.
+  rpc UpdateSiteInfo(UpdateSiteInfoRequest) returns (SiteInfo);
+
+  // Lists registered members on the site.
+  // May be protected or public depending on the privacy policies of the web site.
+  rpc ListMembers(ListMembersRequest) returns (ListMembersResponse);
+
+  // Gets information about a specific member.
+  // May be protected or public depending on the privacy policies of the web site.
+  rpc GetMember(GetMemberRequest) returns (Member);
```

Comment:
What's de difference between `getMember` and `Listmembers`? maybe the permissions are different? if they are the same I would remove `getMember` and add an extra optional parameter in `ListMembersRequest` which is account_id (just like in `getMember`) that would act as a filter so only members with that id would be served. @burdiyan @horacioh @ericvicenti

Reply:
The difference is that List returns multiple users and get returns one. Most of our APIs follow this pattern. Maybe it is enough to have a "filter" in the list method, but I've been always doing it this way, mostly following the [AIP-131 (Get)](https://google.aip.dev/131) and [AIP-132 (List)](https://google.aip.dev/132).

I do think it's clearer and more semantically correct to have these separately though.

For example a Get with a missing ID can clearly return a "not found" error. List method I feel like normally is OK to return an empty collection (and this is a holy war in itself among REST-API aficionados :)). So a List method with an "id" filter should probably return an empty list, which I assume is not what you'd when you only want to get one thing or nothing.

Reply:
Ok, I will leave it as is. On the other hand, who should be able to view member information? everyone or just users with at least editor permissions, @burdiyan ?